packages/Webkul/Customer/src/Http/Controllers/AddressController.php

```diff
@@ -108,31 +109,28 @@ public function edit($id)
      * Edit's the pre-made resource of customer called Address.
      *
      * @param  int  $id
-     * @return \Illuminate\Http\Response
+     * @return \Illuminate\Http\RedirectResponse
      */
     public function update($id, CustomerAddressRequest $request)
     {
         $customer = auth()->guard('customer')->user();
 
-        foreach ($customer->addresses as $address) {
-            if ($id != $address->id) {
-                continue;
-            }
-
-            Event::dispatch('customer.addresses.update.before', $id);
-
-            $customerAddress = $this->customerAddressRepository->update(array_merge($request->all(), [
-                'address1' => implode(PHP_EOL, array_filter(request()->input('address1'))),
-            ]), $id);
+        try {
+            $customer->addresses()->findOrFail($id);
+        } catch (ModelNotFoundException) {
+            session()->flash('warning', trans('shop::app.security-warning'));
+            return redirect()->route('shop.customer.addresses.index');
+        }
```

Comment:
I think this try catch can be reduced by replacing the `findOrFail` to `find` and this just needs one if the case.